pkg/operator/ceph/controller/controller_utils.go

```diff
@@ -138,18 +137,21 @@ func EnforceHostNetwork() bool {
 }
 
 func SetRevisionHistoryLimit(data map[string]string) {
-	strval := k8sutil.GetValue(data, revisionHistoryLimitSettingName, revisionHistoryLimitDefaultValue)
-	if strval != "" {
-		numval, err := strconv.ParseInt(strval, 10, 32)
-		if err != nil {
-			logger.Warningf("failed to parse value %q for %q. assuming default value.", strval, revisionHistoryLimitSettingName)
-			revisionHistoryLimit = nil
-			return
-
-		}
-		limit := int32(numval)
-		revisionHistoryLimit = &limit
+	strval := k8sutil.GetValue(data, revisionHistoryLimitSettingName, "")
+	var limit int32
+	if strval == "" {
+		logger.Debugf("not parsing empty string to int for %q. assuming default value.", revisionHistoryLimitSettingName)
+		revisionHistoryLimit = nil
+		return
+	}
+	numval, err := strconv.ParseInt(strval, 10, 32)
+	if err != nil {
+		logger.Warningf("failed to parse value %q for %q. assuming default value.", strval, revisionHistoryLimitSettingName)
```

Comment:
Let's include the error in the output so users can know what they need to fix 

```suggestion
		logger.Warningf("failed to parse value %q for %q. assuming default value. %v", strval, revisionHistoryLimitSettingName, err)
```